crates/eframe/src/epi/mod.rs

```diff
@@ -383,6 +383,40 @@ pub struct NativeOptions {
     /// Configures wgpu instance/device/adapter/surface creation and renderloop.
     #[cfg(feature = "wgpu")]
     pub wgpu_options: egui_wgpu::WgpuConfiguration,
+
+    /// On Wayland: Application ID for the window.
+    ///
+    /// The application ID is used in several places of the compositor, e.g. for
+    /// grouping windows of the same application. It is also important for
+    /// connecting the configuration of a `.desktop` file with the window, by
+    /// using the application ID as file name. This allows e.g. a proper icon
+    /// handling under Wayland.
+    ///
+    /// See [Waylands XDG shell documentation][xdg-shell] for more information
+    /// on this Wayland-specific option.
+    ///
+    /// [xdg-shell]: https://wayland.app/protocols/xdg-shell#xdg_toplevel:request:set_app_id
```

Comment:
Is there a restriction of what the app id can be? If it can be anything, why not use the app title passed to `run_simple_native`?

Reply:
I think, that application title can be a) dynamic and b) localized which makes it bad choice for such purpose.

Reply:
I agree - I think we should remove `#[cfg(all(feature = "wayland", target_os = "linux"))]` and then also use `app_id` when calling `epi_integration::create_storage`. But I guess I can do that in a follow-up PR

Reply:
Yes, removing the app_id for the storage would probably also be a good fit for this field. Especially, if the window title might become dynamic at some point (e.g. for indicating non-saved files and this kind of things), the `app_id`  could still be static and only set at startup.